types/src/blockchain.rs

```diff
@@ -164,6 +170,21 @@ pub enum BCResponse {
     /// The inner value is `false` if _none_ of the key images were spent.
     KeyImagesSpent(bool),
 
+    /// Response to [`BCReadRequest::CompactChainHistory`].
+    CompactChainHistory {
+        /// A list of blocks IDs in our chain, starting with the most recent block, all the way to the genesis block.
+        ///
+        /// These blocks should be in reverse chronological order, not every block is needed.
+        block_ids: Vec<[u8; 32]>,
+        /// The current cumulative difficulty of the chain.
+        cumulative_difficulty: u128,
+    },
+
+    /// The response for [`BCReadRequest::FindFirstUnknown`].
+    ///
+    /// Contains the index of the first unknown block and its expected height.
+    FindFirstUnknown(Option<(usize, u64)>),
```

Comment:
```suggestion
    /// The response for [`BCReadRequest::FindFirstUnknown`].
    ///
    /// Contains the index of the first unknown block and its expected height.
    ///
    /// This will be [`None`] if all blocks were known.
    FindFirstUnknown(Option<(usize, u64)>),
```